### PR TITLE
feat(components/collapsiblepanel): style update

### PR DESCRIPTION
**Pull request by charlesnguyen**
**What is the problem this PR is trying to solve?**
New guidesline updates: https://company-57688.frontify.com/document/135457#/export-history-v2

Take into account the following changes:
-Left bar with the status color
-Status label must be displayed in black, not colored anymore

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[x] This PR introduces a breaking change**
-The boolean 'selected' used for 'selected CollapsiblePanel' is replaced by the string 'status' in order to customize the panel border
For Example:
selected: true => status = 'selected'
selected: false=> status = ''
-Status component has no longer default export.
import Status from '../Status'; ---> import { Status, getbsStyleFromStatus } from '../Status';

### CODE REVIEW

File: packages/components/src/Status/Status.scss [hunk 1]
```diff
@@ -12,10 +12,28 @@ $tc-status-space-btn-icon-label: 5px !default;
...
+		&.success {
+			color: $brand-success;
+		}
+
+		&.danger {
+			color: $brand-danger;
+		}
+
+		&.muted {
+			color: $brand-muted;
```
Comment (jmfrancois) [inline packages/components/src/Status/Status.scss]: You have created brand-muted on $grey.
I think just color $grey should be enough here.
$brand-xx are bootstrap-sass variable we have changed using our own colors in the bootstrap theme.
I will not block it for that but I would like other opinion

File: packages/components/src/CollapsiblePanel/CollapsiblePanel.component.js [hunk 2]
```diff
@@ -254,13 +229,15 @@ CollapsiblePanel.propTypes = {
 	onSelect: PropTypes.func,
 	onToggle: PropTypes.func,
 	expanded: PropTypes.bool,
-	selected: PropTypes.bool,
+	status: PropTypes.string,
```
Comment (frassinier) [inline packages/components/src/CollapsiblePanel/CollapsiblePanel.component.js]: Breaking changes should be referenced into /BREAKING_CHANGES.MD file :pray:

File: packages/components/src/CollapsiblePanel/CollapsiblePanel.scss [hunk 3]
```diff
@@ -66,6 +66,26 @@ $tc-toggle-color: #555964 !default;
...
+
+	&.danger {
+		border-left: 4px solid $brand-danger;
+		padding-left: 1px;
+	}
+
+	&.muted {
+		border-left: 4px solid $grey;
+		padding-left: 1px;
+	}
```
Comment (frassinier) [inline packages/components/src/CollapsiblePanel/CollapsiblePanel.scss]: Don't Repeat Yourself
```scss
    &.info,
	&.success,
	&.danger,
	&.muted {
		padding-left: 1px;
		border-left: 4px solid;
	}
    
    &.info {
		border-left-color: $brand-info;
	}
 
	&.success {
		border-left-color: $brand-success;
	}
 
	&.danger {
		border-left-color: $brand-danger;
	}
 
	&.muted {
		border-left-color: $grey;
	}
```

File: output/theme.sasslint.txt [hunk 4]
```diff
@@ -28,5 +28,74 @@ src/theme/_tables.scss
...
+  806:26  warning  Hex notation should all be upper case  hex-notation
+  811:32  warning  Hex notation should all be upper case  hex-notation
+  812:35  warning  Hex notation should all be upper case  hex-notation
+  814:26  warning  Hex notation should all be upper case  hex-notation
+  822:29  warning  Hex notation should all be upper case  hex-notation
+  828:14  warning  Hex notation should all be upper case  hex-notation
+  829:11  warning  Hex notation should all be upper case  hex-notation
+  831:13  warning  Hex notation should all be upper case  hex-notation
+  834:10  warning  Hex notation should all be upper case  hex-notation
+  836:20  warning  Hex notation should all be upper case  hex-notation
```
Comment (frassinier) [inline output/theme.sasslint.txt]: Could you change sasslint configuration please?

File: .sass-lint.yml [hunk 5]
```diff
@@ -53,9 +53,7 @@ rules:
     -
       style: 'short'
   hex-notation:
-    - 1
-    -
-      style: 'uppercase'
+    - 0
```
Comment (ncomont) [inline .sass-lint.yml]: You should make a PR on the tools repository (https://github.com/Talend/tools/blob/master/tools-scss-formatting/.sass-lint.yml) prior to this change.

After that, please, make another PR for this change.

File: packages/components/src/Status/Status.component.js [hunk 6]
```diff
@@ -105,5 +97,3 @@ Status.propTypes = {
 Status.defaultProps = {
 	actions: [],
 };
-
-export default Status;
```
Comment (ncomont) [inline packages/components/src/Status/Status.component.js]: Because of this, `Status` component is no more required like that : 

`import Status from 'blah';`

but like that : 

`import { Status } from 'blah';`

So I think this is a breaking change too. What do you think ?

File: .sass-lint.yml (hunk 5 above)
Comment (jmfrancois) [reply .sass-lint.yml]: This is be aligned with perttier ? In that case I auto agree with the PR on tools :D

File: packages/components/src/CollapsiblePanel/CollapsiblePanel.component.js [hunk 7]
```diff
@@ -47,18 +47,11 @@ function renderHeaderItem({ displayMode, className, ...headerItem }, key) {
...
-			return (<Status
-				key={key}
-				actions={adaptActions}
-				{...restStatus}
-				className={css[className]}
-			/>);
+			const adaptActions = actions.map(action => ({
+				...action,
+				onClick: getActionHandler(action.onClick, headerItem),
+			}));
```
Comment (talend-glorieux) [inline packages/components/src/CollapsiblePanel/CollapsiblePanel.component.js]: `adapt` is meaningless. It doesn't describe the transformation applied to the actions. Please find a more descriptive name.

File: packages/components/src/CollapsiblePanel/CollapsiblePanel.component.js [hunk 8]
```diff
@@ -209,40 +192,32 @@ function getTextualContent(content) {
...
-			selected,
-			[css.open]: expanded,
-		});
+	const className = classNames('panel panel-default', css['tc-collapsible-panel'], {
+		[css['default-panel']]: !theme,
+		[css[theme]]: !!theme,
+		[css.open]: expanded,
+		[css[getbsStyleFromStatus(status) || status]]: !!status,
+		status,
+	});
```
Comment (talend-glorieux) [inline packages/components/src/CollapsiblePanel/CollapsiblePanel.component.js]: Lots of complexity for some classes. I can't from a first read be sure of what makes it as class.

File: packages/components/src/Status/Status.component.js [hunk 9]
```diff
@@ -59,28 +59,25 @@ function renderIcon(status, icon, progress) {
...
-function Status(props) {
-	const {
-		status,
-		label,
-		icon,
-		actions,
-		progress,
-	} = props;
+export function Status(props) {
+	const { status, label, icon, actions, progress } = props;
```
Comment (talend-glorieux) [inline packages/components/src/Status/Status.component.js]: Please put that spread inside the function arguments declaration. `props` isn't reused anywhere in the function.

File: output/theme.sasslint.txt [hunk 10]
```diff
@@ -28,5 +28,74 @@ src/theme/_tables.scss
...
+  805:26  warning  Hex notation should all be upper case  hex-notation
+  810:32  warning  Hex notation should all be upper case  hex-notation
+  811:35  warning  Hex notation should all be upper case  hex-notation
+  813:26  warning  Hex notation should all be upper case  hex-notation
+  821:29  warning  Hex notation should all be upper case  hex-notation
+  827:14  warning  Hex notation should all be upper case  hex-notation
+  828:11  warning  Hex notation should all be upper case  hex-notation
+  830:13  warning  Hex notation should all be upper case  hex-notation
+  833:10  warning  Hex notation should all be upper case  hex-notation
+  835:20  warning  Hex notation should all be upper case  hex-notation
```
Comment (talend-glorieux) [inline output/theme.sasslint.txt]: Please fix those errors.

File: packages/components/src/CollapsiblePanel/CollapsiblePanel.component.js (hunk 2 above)
Comment (charlesnguyen) [reply packages/components/src/CollapsiblePanel/CollapsiblePanel.component.js]: done

File: packages/components/src/Status/Status.component.js (hunk 9 above)
Comment (charlesnguyen) [reply packages/components/src/Status/Status.component.js]: done

File: packages/components/src/Status/Status.component.js (hunk 6 above)
Comment (charlesnguyen) [reply packages/components/src/Status/Status.component.js]: done

File: packages/components/src/CollapsiblePanel/CollapsiblePanel.component.js (hunk 7 above)
Comment (charlesnguyen) [reply packages/components/src/CollapsiblePanel/CollapsiblePanel.component.js]: done

File: packages/components/src/CollapsiblePanel/CollapsiblePanel.component.js (hunk 8 above)
Comment (charlesnguyen) [reply packages/components/src/CollapsiblePanel/CollapsiblePanel.component.js]: @talend-glorieux I agree with you but I do not know how to improve it and I do not know why there were so much class.

File: packages/components/BREAKING_CHANGES_LOG.md [hunk 11]
```diff
@@ -1,5 +1,21 @@
 # Breaking changes log
 
+## v0.141.0
```
Comment (frassinier) [inline packages/components/BREAKING_CHANGES_LOG.md]: Should be on https://github.com/Talend/ui/blob/master/BREAKING_CHANGES_LOG.md
